packages/router-core/src/router.ts

```diff
@@ -2643,9 +2641,7 @@ export class RouterCore<
                           const headScripts = headFnContent?.scripts
 
                           const scripts = route.options.scripts?.(assetContext)
-                          const headers = route.options.headers?.({
-                            loaderData,
-                          })
+                          const headers = route.options.headers?.(assetContext)
```

Comment:
are the meta functions down here really executed when an error is thrown? dont we have to run them in the catch block as well?

if my hunch is right, why are your tests not failing?

Reply:
The match.meta is updated here as expected but is there another change you're expecting? I can extend the test to add scripts and links etc too if you want?

Reply:
I was just wondering what happens if the loader throws. it would not reach those lines, so headers/links/ etc would not get executed, right?

Reply:
These lines are executed now because the condition has been updated to include a match.status of `error` or `notFound` 🙂

Reply:
are you on discord? let's discuss there

https://discord.com/invite/WrRKjPJ

Reply:
Please let me know if there's anything you'd like me to do to get this moved forward 🙏🏿